src/test/java/org/sonarsource/sonarlint/ls/mediumtests/ConnectedModeMediumTests.java

```diff
@@ -165,4 +178,18 @@ void analysisConnected_matching_server_issues() throws Exception {
         tuple(2, 6, 2, 11, JAVASCRIPT_S1481, "sonarlint", "Remove the declaration of the unused 'plouf' variable.", DiagnosticSeverity.Warning)));
   }
 
+  @Test
+  void shouldGetServerNamesForConnection() throws Exception {
+    assertLogContains("Enabling notifications for project 'myProject' on connection 'mediumTests'");
+
+    // Trigger a binding update to fetch projects in connected mode storage
+    ExecuteCommandParams updateBindings = new ExecuteCommandParams();
+    updateBindings.setCommand("SonarLint.UpdateAllBindings");
+    lsProxy.getWorkspaceService().executeCommand(updateBindings);
+
+    var params = new GetRemoteProjectsNamesParams(CONNECTION_ID, List.of(PROJECT_KEY1, "unknown"));
+
+    // Update of storage is asynchronous, eventually we get the right data in the storage :)
+    awaitUntilAsserted(() -> assertThat(lsProxy.getRemoteProjectNames(params).get()).containsExactly(Map.entry(PROJECT_KEY1, PROJECT_NAME1)));
+  }
```

Comment:
Do we need tests for when `connectionId` is undefined? Or when the `projectKeys` map is empty? Or when there is no match between the projects response and the projectKeys

Reply:
The `connectionId == null` is indeed a corner case, thanks for noticing!